src/gamepad/gamepad-monitor.vala

```diff
@@ -0,0 +1,63 @@
+// This file is part of GNOME Games. License: GPLv3
+
+/**
+ * This class provides a way to the client to monitor gamepads
+ *
+ * The client interfaces with this class primarily
+ */
+public class LibGamepad.GamepadMonitor : Object {
+	/**
+	 * Emitted when a gamepad is plugged in
+	 * @param  gamepad    The gamepad
+	 */
+	public signal void gamepad_plugged (Gamepad gamepad);
+
+	/**
+	 * Emitted when a gamepad is unplugged
+	 * @param  gamepad    The gamepad
+	 */
+	public signal void gamepad_unplugged (Gamepad gamepad);
+
+	public delegate void GamepadCallback (Gamepad gamepad);
+
+	private static SList <Gamepad> gamepads;
+
+	private RawGamepadMonitor raw_gamepad_monitor;
+
+	public GamepadMonitor () {
+		init_static_if_not ();
+
+		raw_gamepad_monitor = new LinuxRawGamepadMonitor ();
+
+		raw_gamepad_monitor.gamepad_plugged.connect (on_raw_gamepad_plugged);
+
+		raw_gamepad_monitor.foreach_gamepad ((raw_gamepad) => add_gamepad (raw_gamepad));
+	}
+
+	private static void init_static_if_not () {
```

Comment:
Without static state, we wont be able to compare object references across monitors